org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java

```diff
@@ -174,7 +176,12 @@ private void updateProjects(Collection<IProject> projects, long lastWorkspaceSta
 		while (iterator.hasNext()) {
 			IProject project = iterator.next();
 			project.open(monitor);
-			project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+			if (Platform.OS_WIN32.equals(Platform.getOS())) {
+				project.refreshLocal(IResource.DEPTH_ONE, monitor);
```

Comment:
so if I close my client, perform a git checkout/pull on the project(s) I'm working on, which introduces new files, delete others, those changes won't be detected next time the server is restarted right?

@snjeza have you tested it on Windows?

Reply:
You are right. We could use the following patch:
```
diff --git a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java
index 1a034eea..8fd1019b 100644
--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java
@@ -21,6 +21,7 @@ import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
+import org.eclipse.core.internal.resources.Workspace;
 import org.eclipse.core.resources.IContainer;
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IResource;
@@ -206,7 +207,12 @@ public class MavenProjectImporter extends AbstractProjectImporter {
 		while (iterator.hasNext()) {
 			IProject project = iterator.next();
 			project.open(monitor);
-			project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+			Workspace workspace = (Workspace) ResourcesPlugin.getWorkspace();
+			File savedLocation = workspace.getMetaArea().locationFor(project).toFile();
+			File projectFile = project.getLocation().toFile();
+			if (projectFile.lastModified() > savedLocation.lastModified()) {
+				project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+			}
 			if (!needsMavenUpdate(project, lastWorkspaceStateSaved)) {
 				iterator.remove();
 			}
```

Reply:
For original changes, as I tested on Windows, those changes(edit/create/delete/git ops) will be detected on restart during the second refresh and refected on the java dependencies viewer and problems windows. I use this https://github.com/spring-projects/spring-petclinic.git for testing the changes.